docs/ci.md

```diff
@@ -0,0 +1,152 @@
+
+### Examples
+
+Most of our example projects use CircleCI to run `tilt ci`:
+
+<ul>
+  {% for page in site.data.examples %}
+     <li><a href="/{{page.href | escape}}#ci">{{page.title | escape}}</a></li>
+  {% endfor %}
+</ul>
+
+These each use [`ctlptl`](https://ctlptl.dev/) to set up a single-use
+cluster. But if you're considering other options, read on.
+
+## Single-use Kubernetes clusters
+
+If you're using Kubernetes for dev, you'll likely also want to use Kubernetes in CI.
+
+Tilt-team has this problem as well! Our integration test suite uses the latest
+version of Tilt to deploy real sample projects against a real cluster.  So we
+have a lot of experiences with best practices.
+
+Here are some options, with the pros and cons of each:
+
+### Easiest: Cluster on VM-based CI
+
+[`kind`](https://kind.sigs.k8s.io/) is currently the gold standard for running
+Kubernetes in CI. The Kubernetes project itself uses it for testing.
+
+`kind` comes with the ability to run a local registry, so you can push images to
+the registry on `localhost:5000` and pull them from inside `kind`.
+
+Set up a CI pipeline that:
+ 
+1. Creates a VM.
+
+2. Installs all our dependencies, including Docker.
+
+3. Creates a `kind` cluster with a local registry
+   at `localhost:5000`, using [their
+   script](https://kind.sigs.k8s.io/docs/user/local-registry/).
+
+This is the approach we use to test `ctlptl` with both
+`minikube` and `kind`. Here's [the CI
+config](https://github.com/tilt-dev/ctlptl/blob/b6f808a09b05b6cf7aa0b3365e4781d2c23e4851/.circleci/config.yml#L30).
+
+Tilt will auto-detect that the registry is running on `localhost:5000` and push
+images there instead of your prod image registry.
+
+The downside is that most teams are more comfortable managing container
+images than managing VMs. VMs are slower. Upgrading dependencies is more
+heavyweight.
+
+### Recommended: Cluster in Remote Docker-based CI
+
+Many CI environments offer a remote Docker environment outside the container.
+You can run test code in a container that talks to Docker, without the pitfalls
+of running Docker inside Docker.
+
+Set up a CI pipeline that:
+
+1. Create a container with your code.
+
+2. Create [a remote Docker environment](https://circleci.com/docs/2.0/building-docker-images).
+
+3. Start a `kind` cluster with a local registry inside the remote Docker environment.
+
+4. Use `socat` to expose the remote registry and Kubernetes
+   cluster inside the local container.
+
+The `socat` element makes this a bit tricky.
+
+Tilt-team maintains [`ctlptl`](https://ctlptl.dev/), a CLI for declaratively
+setting up local Kubernetes clusters. If you're using `ctlptl`, it will try to
+detect when you have a remote docker environment and set up the `socat`
+forwarding automatically.
+
+If you want to wire it up yourself, check out
+[this Bash script](https://github.com/tilt-dev/kind-local/blob/master/.circleci/with-kind-cluster.sh).
+
+### Not Recommended: Remote Registry
```

Comment:
might put this section earlier, so the "kind|minikube lets you use a local registry!" points have more context/are more impressive